Yank/pipeline.py

```diff
@@ -1112,6 +1111,43 @@ def _generate_molecule(self, molecule_id):
 
         return molecule
 
+    def _generate_residue_name(self, molecule_id):
+        """Generates a residue name for a molecule.
+
+        The function guarantees to not generate twice the same residue name.
+        Purely numeric residue names mess up the pipeline, so we generate
+        residue names of the form YXX, where Y is a letter and X are digits
+        (e.g. A01, A02, ..., Z99).
+
+        The order of the generated residue name is not guaranteed.
+
+        WARNING: The algorithm may fail when new molecules are added to
+        self.molecules after construction. This is not the case right now,
+        but it's good to keep in mind.
+
+        Parameters
+        ----------
+        molecule_id : str
+            The molecule identifier.
+
+        Returns
+        -------
+        residue_name : str
+            A three-character residue name.
+        """
+        # We need to associate a unique number to this molecule, and do
+        # so in such a way that distributing molecule setups over multiple
+        # MPI process still ends up in unique residue names for each molecule.
+        molecule_ids = sorted(self.molecules.keys())
+        n_molecule = molecule_ids.index(molecule_id)
+        assert n_molecule < 2600
```

Comment:
I don't think this will be a problem any time soon, but what happens in the combinatorial space, if someone, for instance, set up a combinatorial molecule which got to `>2600` molecules, would this fail?

Not something we need to fix now by any means, just thinking about the conditions we might see it fail for the first time that a user my not be expecting.

Reply:
It'll keep going through the ascii characters. There are a handful of special symbols that worry me in between uppercase and lowercase letters though, that's why I added the `assert` statement for now. In the future we can skip the next characters and go to lowercase letters to get other 2600 possible resnames.

Reply:
I actually think the assert is a good thing since the current algorithm will generate unique `[A-Z]\d\d` patterns for 2600 molecules, knowing the limit so when/if it is ever hit (probably not any time soon) we know its a problem instead of errors which are not failures (which is way worse)